include/alpaka/core/Sycl.hpp

```diff
@@ -23,6 +23,36 @@
 
 #    include <CL/sycl.hpp>
 
+// if SYCL is enabled with the AMD backend the printf will be killed because of missing compiler support
+#    ifdef __AMDGCN__
+#        include <cstdio> // the define breaks <cstdio> if it is included afterwards
+#        define printf(...)
+#    else
+
+#        ifdef __SYCL_DEVICE_ONLY__
+using AlpakaFormat = char const* [[clang::opencl_constant]];
+#        else
+using AlpakaFormat = char const*;
+#        endif
+
+#        if BOOST_COMP_CLANG
+#            pragma clang diagnostic push
+#            pragma clang diagnostic ignored "-Wgnu-zero-variadic-macro-arguments"
+#        endif
+
+#        define printf(FORMAT, ...)                                                                                   \
+            do                                                                                                        \
+            {                                                                                                         \
+                static auto const format = AlpakaFormat{FORMAT};                                                      \
```

Comment:
```suggestion
                inline static auto const format = AlpakaFormat{FORMAT};                                                      \
```

Reply:
this gives the error: `inline declaration of 'format' not allowed in block scope`

Reply:
Oh sorry, I missed that this macro will be expanded into functions. Ignore this request.